### PR TITLE
fix(calendar): dismiss hover effect when monthsShown is greater than 1

### LINKED ISSUE #3444: Calendar component doesn't dismiss the hover effect when monthsShown={2}
## Current Behavior

When hovering the mouse over the multiple-month calendar (link below), the hover effect doesn't go away when the mouse exits the calendars.

https://baseweb.design/components/datepicker/#calendar-with-multiple-months

Peek 2020-06-29 17-16

## Expected Behavior

The hover effect should hide when the mouse exits the calendar

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.73.2 (including latest on baseweb.design)  |
| React   |   v16.12.0    |
| browser |    Chrome 83     |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

### PR DESCRIPTION
**Pull request by praveenkumar-kalidass**
Fixes #3444 

#### Description
To set highlighted date after mouse leave, isSameMonth and isSameYear check was there previously.

We don't need that because in case of monthsShown={2}, if the mouseLeave event happens on any date on February when selected date is on January, isSameMonth will return false. Hence highlighted text wont return to selected date.

#### Scope
Patch: Bug Fix

### CODE REVIEW

**Comment by chasestarr** [comment]
Thanks for putting up a PR @praveenkumar-outlook! The e2e test below fails. Could you look into that? Here is how to run e2e tests: https://github.com/uber/baseweb/blob/master/TESTING.md

```js
FAIL src/datepicker/__tests__/datepicker-keyboard-nav.e2e.js (11.331s)
--
  | ● Datepicker - keyboard navigation › calendar sets highlighted date appropriately when selecting a new date or navigating around
  |  
  | expect(received).toBe(expected) // Object.is equality
  |  
  | Expected: true
  | Received: false
  |  
  | 249 \|     const isApr1Active = await isActiveEl(page, selectors.apr1);
  | 250 \|     // the first calendar's day is focused
  | > 251 \|     expect(isApr1Active).toBe(true);
  | \|                          ^
  | 252 \|
  | 253 \|     // Navigate back to March with a selected range in it
  | 254 \|     // and verify that this time it highlights the beginning date of
  |  
  | at _callee2$ (src/datepicker/__tests__/datepicker-keyboard-nav.e2e.js:251:26)
  | at tryCatch (node_modules/regenerator-runtime/runtime.js:45:40)
  | at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:271:22)
  | at Generator.prototype.<computed> [as next] (node_modules/regenerator-runtime/runtime.js:97:21)
  | at tryCatch (node_modules/regenerator-runtime/runtime.js:45:40)
  | at invoke (node_modules/regenerator-runtime/runtime.js:135:20)
  | at node_modules/regenerator-runtime/runtime.js:145:13
```

File: src/datepicker/calendar.js [hunk 1]
```diff
@@ -397,11 +397,7 @@ export default class Calendar<T = Date> extends React.Component<
     const {value} = this.props;
     const selected = this.getSingleDate(value);
```
Comment (nadiia) [inline src/datepicker/calendar.js:398]:
```suggestion
    const [startDate, endDate] = [].concat(this.props.value);
```

File: src/datepicker/calendar.js [hunk 2]
```diff
@@ -397,11 +397,7 @@ export default class Calendar<T = Date> extends React.Component<
...
     let nextState;
-    if (
-      selected &&
```
Comment (nadiia) [inline src/datepicker/calendar.js:401]: I think the fix here would be to check if it's a start or end date. See the code change suggestions.

File: src/datepicker/calendar.js [hunk 3]
```diff
@@ -397,11 +397,7 @@ export default class Calendar<T = Date> extends React.Component<
...
-      this.dateHelpers.isSameMonth(selected, date) &&
-      this.dateHelpers.isSameYear(selected, date)
-    ) {
+    if (selected) {
```
Comment (nadiia) [inline src/datepicker/calendar.js:400]:
```suggestion
    if (
      startDate &&
      this.dateHelpers.isSameMonth(startDate, date) &&	
      this.dateHelpers.isSameYear(startDate, date)
    ) {
```

File: src/datepicker/calendar.js [hunk 4]
```diff
@@ -397,11 +397,7 @@ export default class Calendar<T = Date> extends React.Component<
...
       nextState = {highlightedDate: selected};
```
Comment (nadiia) [inline src/datepicker/calendar.js:401]:
```suggestion
      nextState = {highlightedDate: startDate};
```

File: src/datepicker/calendar.js [hunk 5]
```diff
@@ -397,11 +397,7 @@ export default class Calendar<T = Date> extends React.Component<
...
     } else {
```
Comment (nadiia) [inline src/datepicker/calendar.js:402]:
```suggestion
    } else if (
      endDate &&
      this.dateHelpers.isSameMonth(endDate, date) &&	
      this.dateHelpers.isSameYear(endDate, date)
    ) {
      nextState = {highlightedDate: endDate};
    } else {
```

File: src/datepicker/calendar.js (hunk 2 above)
Comment (praveenkumar-outlook) [reply src/datepicker/calendar.js:401]: If this change is made, then for case of monthsShown={3} when the mouseleave happens outside range, the date wont be highlighted, is that okay?
Screenshot 2020-08-08 at 11 21 38 AM

**Comment by williamernest** [comment]
Resolved with #4336 4336

**Pull request closed without merging**